### PR TITLE
Screenshots path in report fixed for tests in quarantine mode (closes #746)

### LINKED ISSUE #746: Fix screenshots path in report for tests in quarantine mode
Due to https://github.com/DevExpress/testcafe/pull/732#issuecomment-242022126
Now if we run test in quarantine mode in screenshotBase dir will be created three dirs:
```
-{task-start-date}
    -test-1
        -....
    -test-1-1
        -....
    -test-1-2
        -....
```
And in report will be specified last path: `{basePath}\{task-start-date}\test-1-2`.
We have two ideas to fix it:
1) specified several paths in report
2) or use for quarantine mode different way to save screenshots:
```
-{task-start-date}
    -test-1
        -run-1
            -...
        -run-2
            -...
        -run-3
            -...
```
for report we will use `{basePath}\{task-start-date}\test-1` path.

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @inikulin  @AlexanderMoskovkin

### CODE REVIEW

File: src/runner/browser-job.js [hunk 1]
```diff
@@ -85,7 +86,8 @@ export default class BrowserJob extends EventEmitter {
 
     _createTestRun (test, testIndex, quarantineAttemptNum) {
         var TestRunCtor        = test.isLegacy ? LegacyTestRun : TestRun;
-        var screenshotCapturer = this.screenshots.createCapturerFor(test, testIndex, quarantineAttemptNum, this.browserConnection);
+        var screenshotCapturer = this.screenshots.createCapturerFor(test, testIndex,
+            this.opts.quarantineMode ? quarantineAttemptNum : null, this.browserConnection);
```
Comment (inikulin) [inline src/runner/browser-job.js]: Let's keep it single line, just refactor `this.opts.quarantineMode ? quarantineAttemptNum : null` into var

File: src/screenshots/capturer.js [hunk 2]
```diff
@@ -60,14 +56,18 @@ export default class Capturer {
             screenshotPath      = joinPath(this.baseScreenshotsPath, Capturer._correctFilePath(customPath));
         }
         else {
-            var path = joinPath(this.baseScreenshotsPath, this.baseDirName, this.testDirName);
+            var testDirName       = `test-${this.testIndex}`;
+            var path              = joinPath(this.baseScreenshotsPath, this.baseDirName, testDirName);
+            var pathForScreenshot = this.quarantineAttemptNum !== null ?
+                                    joinPath(path, `run-${this.quarantineAttemptNum}`) :
+                                    path;
```
Comment (inikulin) [inline src/screenshots/capturer.js]: Make it single line

Comment (inikulin) [inline src/screenshots/capturer.js]: I don't like `pathForScreenshot` name, BTW. `path` is path for screenshot too. Just get rid of ternary here and make conditional assignment of `joinPath(path, `run-${this.quarantineAttemptNum}`)`

**Comment by inikulin** [comment]
\r-

**Comment by inikulin** [comment]
@helen-dikareva What about https://github.com/DevExpress/testcafe/pull/750#discussion_r77179368?

**Comment by helen-dikareva** [comment]
Sorry, I didn't notice it

**Comment by inikulin** [comment]
lgtm

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**